### ISSUE TITLE
App crashes while creating an password protected sharing link

### ISSUE THREAD

**Issue by Z3NOX** [nextcloud/android#8102]
### Steps to reproduce
1. Open the app, choose any file
2. click 'details', 'share', add a new public link
3. Open the menu to add a password, enter password - crash

### Expected behaviour
It should work without crashing

### Actual behaviour
It crashes

### Can you reproduce this problem on https://try.nextcloud.com?
- actually I am unable to create a test account there, sorry.

### Environment data
Android version:

Device model: 
- Pixel 4a

Stock or customized system:
- CalyxOS

Nextcloud app version:
- 3.15.1

Nextcloud server version:
- 20.0.0.4

Reverse proxy:

#### Web server error log
```
Insert your webserver log here
```

#### Nextcloud app error
```
************ CAUSE OF ERROR ************

java.lang.RuntimeException: Accessing result data after operation failed!
	at com.owncloud.android.lib.common.operations.RemoteOperationResult.getData(RemoteOperationResult.java:486)
	at com.owncloud.android.utils.ErrorMessageAdapter.getMessageForUpdateShareOperations(ErrorMessageAdapter.java:204)
	at com.owncloud.android.utils.ErrorMessageAdapter.getMessageForResultAndOperation(ErrorMessageAdapter.java:152)
	at com.owncloud.android.utils.ErrorMessageAdapter.getErrorCauseMessage(ErrorMessageAdapter.java:79)
	at com.owncloud.android.ui.activity.FileActivity.onRemoteOperationFinish(FileActivity.java:380)
	at com.owncloud.android.ui.activity.FileDisplayActivity.onRemoteOperationFinish(FileDisplayActivity.java:1704)
	at com.owncloud.android.services.OperationsService.lambda$dispatchResultToOperationListeners$0(OperationsService.java:734)
	at com.owncloud.android.services.-$$Lambda$OperationsService$KsB1LbjHjlEBSp0h7XO24czs-OQ.run(Unknown Source:6)
	at android.os.Handler.handleCallback(Handler.java:938)
	at android.os.Handler.dispatchMessage(Handler.java:99)
	at android.os.Looper.loop(Looper.java:223)
	at android.app.ActivityThread.main(ActivityThread.java:7660)
	at java.lang.reflect.Method.invoke(Native Method)
	at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:592)
	at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:947)

************ APP INFORMATION ************
ID: com.nextcloud.client
Version: 30150090
Build flavor: generic

************ DEVICE INFORMATION ************
Brand: Android
Device: sunfish
Model: Pixel 4a
Product: calyx_sunfish

************ FIRMWARE ************
SDK: 30
Release: 11
Incremental: 2021.02.02.18
```
**NOTE:** Be super sure to remove sensitive data like passwords, note that everybody can look here! You can use the Issue Template application to prefill some of the required information: https://apps.nextcloud.com/apps/issuetemplate

**Comment by AndyScherzinger** [comment]
@Z3NOX afaik this has been fixed by @tobiasKaminsky and is available in 3.15.1 RC1

**Comment by tobiasKaminsky** [comment]
Please re-open if this happens with latest beta.
